runtime/engine/src/main/java/io/aklivity/zilla/runtime/engine/Engine.java

```diff
@@ -197,24 +218,23 @@ public long counter(
         return counter.applyAsLong(name);
     }
 
-    public Future<Void> start()
+    public Future<Void> start() throws ExecutionException, InterruptedException
     {
         for (AgentRunner runner : runners)
         {
             AgentRunner.startOnThread(runner, Thread::new);
         }
-
-        this.configureRef = commonPool().submit(configure);
-
-        return configureRef;
+        reconfigure();
```

Comment:
I think we should call the initial configuration here instead of the WatcherTask. Tried it that way, and since we won't wait for the WatcherTask to finish (long running task), it can happen (and happened as well) that the Zilla starts the stopping process before the configuration is actually finished. A thing that we can do here instead of this, is to wait for the register CountDownLatch. Let me know if you prefer that approach.